libraries/AP_Proximity/AP_Proximity_DroneCAN.cpp

```diff
@@ -0,0 +1,198 @@
+    measurement_listener = new uavcan::Subscriber<ardupilot::equipment::proximity_sensor::Proximity, MeasurementCb>(*node);
+    // Register method to handle incoming RangeFinder measurement
+    const int measurement_listener_res = measurement_listener->start(MeasurementCb(ap_uavcan, &handle_measurement));
+    if (measurement_listener_res < 0) {
+        AP_HAL::panic("DroneCAN Proximity subscriber start problem\n\r");
+        return;
+    }
+}
+
+//Method to find the backend relating to the node id
+AP_Proximity_DroneCAN* AP_Proximity_DroneCAN::get_uavcan_backend(AP_UAVCAN* ap_uavcan, uint8_t node_id, uint8_t address, bool create_new)
+{
+    if (ap_uavcan == nullptr) {
+        return nullptr;
+    }
+    AP_Proximity_DroneCAN* driver = nullptr;
+    AP_Proximity &prx = *AP::proximity();
+    //Scan through the Rangefinder params to find UAVCAN RFND with matching address.
+    for (uint8_t i = 0; i < PROXIMITY_MAX_INSTANCES; i++) {
+        if ((AP_Proximity::Type)prx.params[i].type.get() == AP_Proximity::Type::UAVCAN &&
+            prx.params[i].address == address) {
+            driver = (AP_Proximity_DroneCAN*)prx.drivers[i];
+        }
+        //Double check if the driver was initialised as UAVCAN Type
+        if (driver != nullptr && (driver->_backend_type == AP_Proximity::Type::UAVCAN)) {
+            if (driver->_ap_uavcan == ap_uavcan &&
+                driver->_node_id == node_id) {
+                return driver;
+            } else {
+                //we found a possible duplicate addressed sensor
+                //we return nothing in such scenario
+                return nullptr;
+            }
+        }
+    }
+
+    if (create_new) {
+        for (uint8_t i = 0; i < PROXIMITY_MAX_INSTANCES; i++) {
+            if ((AP_Proximity::Type)prx.params[i].type.get() == AP_Proximity::Type::UAVCAN &&
+                prx.params[i].address == address) {
+                WITH_SEMAPHORE(prx.detect_sem);
+                if (prx.drivers[i] != nullptr) {
+                    //we probably initialised this driver as something else, reboot is required for setting
+                    //it up as UAVCAN type
+                    return nullptr;
+                }
+                prx.drivers[i] = new AP_Proximity_DroneCAN(prx, prx.state[i], prx.params[i]);
+                driver = (AP_Proximity_DroneCAN*)prx.drivers[i];
+                if (driver == nullptr) {
+                    break;
+                }
+                gcs().send_text(MAV_SEVERITY_INFO, "Prx[%u]: added UAVCAN node %u addr %u",
+                                unsigned(i), unsigned(node_id), unsigned(address));
+
+                if (is_zero(prx.params[i].max_m) && is_zero(prx.params[i].min_m)) {
+                    // GCS reporting will be incorrect if min/max are not set
+                    gcs().send_text(MAV_SEVERITY_CRITICAL, "Configure PRX%u_MIN and PRX%u_MAX",
```

Comment:
I suppose this message is unlikely to be seen.  An alternative would be to make it appear when prearm_healthy() is called somehow.  This is a bit optional though.

Reply:
I agree, but I think I will do this as part of another PR perhaps